tools/crates-validator/src/lib.rs

```diff
@@ -361,18 +419,25 @@ impl Validator {
         // Generate the main.rs file
 
         let mut vars = BTreeMap::new();
-        vars.insert("no_std_line", if profile.std { "" } else { "#![no_std]" });
+        vars.insert(
+            "no_std_line",
+            if profile.settings.std {
+                ""
+            } else {
+                "#![no_std]"
+            },
+        );
```

Comment:
Maybe all these if else to string creation could be changed to .then methods for Boolean and Option?